lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
I think the root of the issue is that we can't actually validate our code without a change to the schema. to merge this PR would just be guessing what the schema is going to be in the meantime.  It's a pretty good guess, but I'm sure everyone using JSON API in AMS would feel more comfortable if we were inline with the schema.